cpp/cmake_modules/SetupCxxFlags.cmake

```diff
@@ -24,8 +24,11 @@ CHECK_CXX_COMPILER_FLAG("-msse3" CXX_SUPPORTS_SSE3)
 CHECK_CXX_COMPILER_FLAG("-maltivec" CXX_SUPPORTS_ALTIVEC)
 
 # compiler flags that are common across debug/release builds
-#  - Wall: Enable all warnings.
-set(CXX_COMMON_FLAGS "-std=c++11 -Wall")
+#  - Wall: Enable all warnings. Not used for MSVC,
+# because cause build failure by too much warnings.
+if (NOT MSVC)
+    set(CXX_COMMON_FLAGS "-std=c++11 -Wall")
```

Comment:
I have found that nmake complained that -std=c++11 command is not recognized. According to https://blogs.msdn.microsoft.com/vcblog/2016/06/07/standards-version-switches-in-the-compiler/ "Note that there is no plan to add a C++11 switch.". It seems MS recently added /std:c++14 switch to VS 2015 nmake distribution, but all previous C++ standards implemented features are enabled by default and there are no command line switches, etc. Pretty weird.